src/go/rpk/pkg/cli/cluster/partitions/toggle.go

```diff
@@ -0,0 +1,243 @@
+provided, rpk will assume 'kafka'. If the topic is not provided in the flag, rpk
+will use the topic provided as an argument to this command.
+
+DISABLED PARTITIONS
+
+Disabling a partition in Redpanda involves prohibiting any data consumption or
+production to and from it. All internal processes associated with the partition
+are stopped, and it remains unloaded during system startup. This measure aims to
+maintain cluster health by preventing issues caused by specific corrupted
+partitions that may lead to Redpanda crashes. Although the data remains stored
+on disk, Redpanda ceases interaction with the disabled partitions to ensure
+system stability.
+
+EXAMPLES
+
+Disable all partitions in topic 'foo'
+    rpk cluster partitions disable foo --all
+
+Disable partitions 1,2 and 3 of topic 'bar' in the namespace 'internal'
+    rpk cluster partitions disable internal/bar --partitions 1,2,3
+
+Disable partition 1, and 2 of topic 'foo', and partition 5 of topic 'bar' in the 
+'internal' namespace' 
+    rpk cluster partitions disable -p foo/1,2 -p internal/bar/5
+`,
+		Args: cobra.MaximumNArgs(1),
+		Run: func(cmd *cobra.Command, topicArg []string) {
+			if !all && partitions == nil {
+				fmt.Println("Please select either specific partitions to disable (--partitions) or select all (--all)")
+				cmd.Help()
+				os.Exit(1)
+			}
+			if all && len(topicArg) == 0 {
+				fmt.Println("You must select a topic")
+				cmd.Help()
+				os.Exit(1)
+			}
+			p, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+			out.CheckExitCloudAdmin(p)
+
+			cl, err := adminapi.NewClient(fs, p)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			err = runToggle(cmd.Context(), cl, all, topicArg, partitions, "disable")
+			out.MaybeDieErr(err)
+			if all {
+				fmt.Printf("Successfully disabled all partitions in topic %v\n", topicArg[0])
+			} else {
+				fmt.Printf("Successfully disabled the partitions %v\n", partitions)
+			}
+		},
+	}
+	cmd.Flags().StringArrayVarP(&partitions, "partitions", "p", nil, "Comma-separated list of partitions you want to disable. Use --help for additional information")
+	cmd.Flags().BoolVarP(&all, "all", "a", false, "If true, disable all partitions")
+
+	cmd.MarkFlagsMutuallyExclusive("partitions", "all")
+	return cmd
+}
+
+func runToggle(ctx context.Context, cl *adminapi.AdminAPI, all bool, topicArg, partitionFlag []string, verb string) error {
+	isDisable := verb == "disable"
+	if all {
+		ns, topicName := nsTopic(topicArg[0])
+		err := cl.ToggleAllTopicPartitions(ctx, isDisable, ns, topicName)
+		if err != nil {
+			return fmt.Errorf("failed to %v all partitions in topic %v/%v: %v", verb, ns, topicName, err)
+		}
+		return nil
+	}
+	g, egCtx := errgroup.WithContext(ctx)
+	for _, ntp := range partitionFlag {
+		ns, topicName, partitions, err := parsePartition(ntp)
+		if err != nil {
+			return err
+		}
+		if topicName != "" && len(topicArg) > 0 && topicArg[0] != "" {
```

Comment:
Is it possible for len(topicArg) to be greater than zero, but also for topicArg[0] to be empty? I wonder if this can just be `if topicName != "" && len(topicArg) > 0 {`

Reply:
Yes, this came up during testing, cobra will parse:

```
rpk cluster partitions enable ""
```
As topicArgs to be a `[]string{""}`, so, length 1, but empty string.